pkg/apis/kops/cluster.go

```diff
@@ -148,11 +148,31 @@ type ClusterSpec struct {
 	// This API component is under construction, will remove this comment
 	// once this API is fully functional.
 	Assets *Assets `json:"assets,omitempty"`
-
 	// IAM field adds control over the IAM security policies applied to resources
 	IAM *IAMSpec `json:"iam,omitempty"`
 	// EncryptionConfig controls if encryption is enabled
 	EncryptionConfig *bool `json:"encryptionConfig,omitempty"`
+	// SecurityGroups allows for use an existing custom cloud security group for the instances.
+	// One example is to specify the name of an AWS security group for master and another for the nodes.
+	SecurityGroups *SecurityGroups `json:"securityGroups,omitempty"`
+}
+
+// Security groups for master(s) and node(s)
+type SecurityGroups struct {
+	// Master is the name of the sg to use for the master
+	Master *string `json:"master,omitempty"`
+
+	// Node is the name of the sg to use for the nodes
+	Node *string `json:"node,omitempty"`
+
+	// Bastion is the name of the sg to use for the bastion
+	Bastion *string `json:"bastion,omitempty"`
+
+	// BastionELB is the name of the sg to use for the bastion elb
+	BastionELB *string `json:"bastionELB,omitempty"`
+
+	// ApiELB is the name of the sg to use for the api elb
+	ApiELB *string `json:"apiELB,omitempty"`
```

Comment:
Wondering if this should be on the API LoadBalancerAccessSpec .  I think probably yes.  Not sure about whether we would then apply the same reasoning to the other fields here...

Reply:
It does make sense to me as well.  I was trying for grouping and simplicity.

We have lots of options, but I was thinking putting all of the base security groups in the same place would make more sense.  I would literally have to document two places instead of one.  That was my approach.

You want me to move?  Happy to.

Reply:
Going to move ... otherwise, it will be confusing.